src/init.cpp

```diff
@@ -307,6 +308,14 @@ static void HandleSIGHUP(int)
 {
     fReopenDebugLog = true;
 }
+#else
+static BOOL WINAPI consoleCtrlHandler(DWORD dwCtrlType)
+{
+    fRequestShutdown = true;
+    Sleep(-1);
```

Comment:
What does sleeping a negative amount of time do?

Reply:
For sleeping as long as possible and waiting for main thread quit.

Reply:
I've checked MSDN docs: https://msdn.microsoft.com/en-us/library/windows/desktop/ms686298(v=vs.85).aspx  and the parameter is a DWORD (unsigned) to -1 is not a valid value.
It seems that there is a constant `INFINITE` to wait infinitely, if that is what is the intent?

Reply:
INFINITE is 0xFFFFFFFF, so it might be the same. But to be safe, I would change it to the micro.